chia/wallet/did_wallet/did_wallet.py

```diff
@@ -605,10 +608,20 @@ async def transfer_did(self, puzzle_hash: bytes32, fee: uint64) -> TransactionRe
         assert coins is not None
         coin = coins.pop()
 
+        new_did_puz = did_wallet_puzzles.create_innerpuz(
+            new_puzhash,
+            0,
+            0,
+            self.did_info.origin_coin.name(),
+            self.did_info.metadata,
+        )
+        p2_solution = p2_solution = self.standard_wallet.make_solution(
```

Comment:
I think we need to reveal the recovery list in the transfer case. How doesn't the recipient know the recovery list in this approach?

Reply:
We also need to modify the create_new_did_wallet_from_coin_spend(), since the curried parameters and solution are changed.